packages/contractkit/src/utils/azure-key-vault-client.ts

```diff
@@ -151,18 +151,12 @@ export class AzureKeyVaultClient {
   ): number {
     for (let i = 0; i < 4; i++) {
       const compressed = false
-      try {
-        const recoveredPublicKeyByteArr = ecdsaRecover(signature, i, hash, compressed)
-        const publicKeyBuff = Buffer.from(recoveredPublicKeyByteArr)
-        const recoveredPublicKey = AzureKeyVaultClient.bufferToBigNumber(publicKeyBuff)
-        debug('Recovered key: ' + recoveredPublicKey)
-        if (publicKey.eq(recoveredPublicKey)) {
-          return i
-        }
-      } catch (e) {
-        if (e.message === 'Public key could not be recover') {
```

Comment:
I'd still like to keep this try/catch here since the user may be attempting to sign using a non-ECDSA key and this message might help them locate the issue.

Reply:
try catch as it was before was silently swallowing error messages because it wasn't re-throwing errors when message didnt match this constant. 
I think it is cleaner without the try/catch and 'ecdsaRecover' will show up in stack trace anyways for anyone who would be debugging issues. It is pretty clear that it is talking about ecdsa keys.

Reply:
also anyone using this, would have to have at least some understanding of what kind of key to use anyways, because ECDSA isn't explicit. There are different types of ECDSA keys, and user must be using specifically 'secp256k1' version, others wont work.

Reply:
That's fair. We can also keep the original logic and just rethrow the error in the case it doesn't match this message. I'll leave it up to you.